curriculum/challenges/english/08-coding-interview-prep/rosetta-code/zhang-suen-thinning-algorithm.english.md

```diff
@@ -44,49 +44,52 @@ It produces the thinned output:
                        ###                          ###
 
-                                                           </pre>
+</pre>
+
 <h2>Algorithm</h2>
 Assume black pixels are one and white pixels zero, and that the input image is a rectangular N by M array of ones and zeroes.
 The algorithm operates on all black pixels P1 that can have eight neighbours. The neighbours are, in order, arranged as:
-<table border="1">
-  <tr><td>P9</td><td>P2</td><td>P3</td></tr>
-  <tr><td>P8</td><td><b>P1</b></td><td>P4</td></tr>
-  <tr><td>P7</td><td>P6</td><td>P5</td></tr>
-</table>
-Obviously the boundary pixels of the image cannot have the full eight neighbours.
-
-    Define $A(P1)$ = the number of transitions from white to black, (0 -> 1) in the sequence P2,P3,P4,P5,P6,P7,P8,P9,P2. (Note the extra P2 at the end - it is circular).
 
+<table border="3">	
+  <tr><td style="text-align: center;">P9</td><td style="text-align: center;">P2</td><td style="text-align: center;">P3</td></tr>	
+  <tr><td style="text-align: center;">P8</td><td style="text-align: center;"><b>P1</b></td><td style="text-align: center;">P4</td></tr>	
+  <tr><td style="text-align: center;">P7</td><td style="text-align: center;">P6</td><td style="text-align: center;">P5</td></tr>	
+</table>
 
-    Define $B(P1)$ = the number of black pixel neighbours of P1. ( = sum(P2 .. P9) )
+Obviously the boundary pixels of the image cannot have the full eight neighbours.
+  <ul>
+    <li>Define $A(P1)$ = the number of transitions from white to black, (0 -> 1) in the sequence P2, P3, P4, P5, P6, P7, P8, P9, P2. (Note the extra P2 at the end - it is circular).</li>
+    <li>Define $B(P1)$ = the number of black pixel neighbours of P1. ( = sum(P2 .. P9) )</li>
+  </ul>
 
 <h3>Step 1:</h3>
 All pixels are tested and pixels satisfying all the following conditions (simultaneously) are just noted at this stage.
-  (0) The pixel is black and has eight neighbours
-  (1) $2 <= B(P1) <= 6$
-  (2) $A(P1) = 1$
-  (3) At least one of P2 and P4 and P6 is white
-  (4) At least one of P4 and P6 and P8 is white
+  <ol>
+    <li>The pixel is black and has eight neighbours</li>
+    <li>$2 <= B(P1) <= 6$</li>
+    <li>$A(P1) = 1$</li>
+    <li>At least one of P2 and P4 and P6 is white</li>
+    <li>At least one of P4 and P6 and P8 is white</li>
+  </ol>
 After iterating over the image and collecting all the pixels satisfying all step 1 conditions, all these condition satisfying pixels are set to white.
+
 <h3>Step 2:</h3>
 All pixels are again tested and pixels satisfying all the following conditions are just noted at this stage.
-  (0) The pixel is black and has eight neighbours
-  (1) $2 <= B(P1) <= 6$
-  (2) $A(P1) = 1$
-  (3) At least one of P2 and P4 and '''P8''' is white
-  (4) At least one of '''P2''' and P6 and P8 is white
+  <ol>
+    <li>The pixel is black and has eight neighbours</li>
+    <li>$2 <= B(P1) <= 6$</li>
+    <li>$A(P1) = 1$</li>
+    <li>At least one of <b>P2, P4 and P8</b> is white</li>
```

Comment:
@thecodingaviator, good catch with the commas here.

Also, I'm not sure why, but on https://rosettacode.org/wiki/Zhang-Suen_thinning_algorithm, just P8 and P2 are made bold. Maybe it was a typo? What do you think?

Reply:
Yeah, I guess it was a mistake